proteuslib/property_models/tests/property_test_harness.py

```diff
@@ -0,0 +1,532 @@
+
+        # check method naming convention
+        for v_str in metadata.keys():
+            if v_str in m.state_vars_list:
+                assert metadata[v_str]['method'] is None
+            else:
+                assert metadata[v_str]['method'] == '_' + v_str
+
+    @pytest.mark.unit
+    def test_build(self, frame):
+        m = frame
+
+        # properties and state variables
+        prop_str_list = []
+        for obj_str, (obj_type, obj_units) in m.stateblock_obj_dict.items():
+            if obj_type is Expression or obj_type is Var:
+                prop_str_list.append(obj_str)
+
+        # check that the testing property list matches metadata
+        metadata = m.fs.properties.get_metadata().properties
+        assert len(metadata) == len(prop_str_list)
+        for v_str in metadata.keys():
+            assert v_str in prop_str_list
+
+        # check that properties are not built if not demanded
+        for obj_str in prop_str_list:
+            if obj_str not in m.state_vars_list:
+                assert not m.fs.stream[0].is_property_constructed(obj_str)
+
+        # check that properties are built on demand
+        for obj_str in prop_str_list:
+            assert hasattr(m.fs.stream[0], obj_str)
+
+        # add constraints for on demand variables to testing dictionary
+        for obj_str in prop_str_list:
+            if (m.stateblock_obj_dict[obj_str][0] is Var
+                    and obj_str not in m.state_vars_list):
+                m.stateblock_obj_dict['eq_' + obj_str] = (Constraint, None)
+
+        self.check_block_obj(m.fs.stream[0], m.stateblock_obj_dict)
+        self.check_block_obj_coverage(m.fs.stream[0], m.stateblock_obj_dict)
+
+    @pytest.mark.unit
+    def test_statistics(self, frame):
+        m = frame
+
+        assert number_variables(m) == m.statistics_dict['number_variables']
+        assert number_total_constraints(m) == m.statistics_dict['number_total_constraints']
+        assert number_unused_variables(m) == m.statistics_dict['number_unused_variables']
+        assert degrees_of_freedom(m) == m.statistics_dict['default_degrees_of_freedom']
+
+    @pytest.mark.unit
+    def test_general_methods(self, frame):
+        m = frame
+
+        # TODO: update for dynamic compatible property packages
+        # general methods requirements
+        required_general_methods_list = ['get_material_flow_terms',
+                                         'get_enthalpy_flow_terms']
+        tested_general_methods_list = []
+        for (method_name, ind) in m.general_methods_dict.keys():
+            if method_name not in tested_general_methods_list:
+                tested_general_methods_list.append(method_name)
+        for i in required_general_methods_list:
+            assert i in tested_general_methods_list
+
+        for (m_str, m_ind), (v_str, v_ind) in m.general_methods_dict.items():
+            assert hasattr(m.fs.stream[0], m_str)
```

Comment:
What does this test actually do? Also, the name `general_method` looks like it is a reserved name - I think you should choose something else.